src/hostapi/wasapi/pa_win_wasapi.c

```diff
@@ -1414,6 +1419,36 @@ static MixMonoToStereoF GetMonoToStereoMixer(const WAVEFORMATEXTENSIBLE *fmtext,
     return NULL;
 }
 
+// ------------------------------------------------------------------------------------------
+static HRESULT ReallocateMonoMixerBuffer(PaWasapiSubStream *subStream, UINT32 frames)
+{
+    assert(subStream->monoMixer != NULL);
+
+    // If not provided then use max buffer size of WASAPI device
+    if (frames == 0)
+        frames = subStream->bufferSize;
+
+    assert(frames != 0);
+
+    // Expand buffer if necessary
+    UINT32 monoBufferSize = frames * subStream->wavexu.ext.Format.nBlockAlign;
+    if (monoBufferSize > subStream->monoBufferSize)
+    {
+        subStream->monoBuffer = PaWasapi_ReallocateMemory(subStream->monoBuffer, monoBufferSize);
+        if (subStream->monoBuffer == NULL)
+        {
+            subStream->monoBufferSize = 0;
+            return E_OUTOFMEMORY;
+        }
+
+        subStream->monoBufferSize = monoBufferSize;
+    }
+
+    assert(subStream->monoBuffer != NULL);
```

Comment:
It is a sanity check of the logic, in case buffer is not allocated at all due to `subStream->wavexu.ext.Format.nBlockAlign`. I modified assertion in the subsequent commit.